lib/curl.dart

```diff
@@ -76,11 +78,9 @@ class CurlClient implements Client {
       }
       args.add(socksHostPort);
     }
-
-    final method = request.method ?? 'GET';
     args.addAll(['-X', method.toUpperCase()]);
-    // TODO: add data processing, e.g. for strings:
-    // if (data != null) args.addAll(['--data', data]);
+    // --data parameter added in GET and POST Method. If method is 'GET', body may no have any effect in the request. UTF-8 encoding setted as default.
+    if (request.body != null) args.addAll(['--data', utf8.decode(request.body)]);
```

Comment:
I think this needs to have a type check on request.body (it can be `String`, `List<int>`, even `File` per the interface specification). It should be updated in a future version to be only one of them though, but for now, we need to deal with it :(
You can also guard this by `if (request.body is! List<int>) throw Exception();` or similar.

Reply:
Hello, Changelog record added as 1.4.1+1, version changed as the same and basic body type check implemented too.
Please check!

Reply:
I'll add a null-check after this is merged, but otherwise looks good, thank you!